### DISCUSSION TITLE
Inline Image Previews?

### DISCUSSION THREAD

**Discussion by alexispurslane** [stoatchat/discussions#527] [Feature Suggestions]
I'm self-hosting a Revolt instance, and one of the first things I noticed is that, at least in the web app/PWA, when someone uploads an image it displays the image filename and a download button instead of the image itself, which is inconvenient. Would this be relatively easy to fix, or is there some structural thing blocking it?

**Comment by insertish** [comment]
This seems like a bug, images are only displayed when their metadata can be read.
Currently only JPEG, PNG, GIF, and ~~WEBP~~ (in theory but it doesn't seem to decode correctly, although I just had a hint of what could be going wrong) are supported though.

Expected behaviour:
<img width="534" alt="Screenshot 2023-04-22 at 13 14 11" src="https://user-images.githubusercontent.com/38285861/233784060-693998bd-bfe3-4495-87b1-b5138215ed08.png">

**Reply by alexispurslane** [reply]
Hmm, okay. I'll try checking the devtools to see if there are any API endpoints or whatever it isn't able to find or something. I believe the images should be a type it's able to display.

**Reply by alexispurslane** [reply]
Okay, it isn't able to display images uploaded from the PWA on my phone inline. Images uploaded from my laptop do display inline just fine. But there aren't any errors in the console.
